android-sdk-restclient-test/src/test/java/healthvault/microsoft/com/android_sdk_restclient_test/ActionPlanTest.java

```diff
@@ -0,0 +1,86 @@
+package healthvault.microsoft.com.android_sdk_restclient_test;
+
+import android.content.Context;
+
+import org.junit.Test;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URISyntaxException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.reflect.TypeToken;
+import com.microsoft.hsg.Connection;
+import com.microsoft.hsg.android.simplexml.ConnectionFactory;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultClient;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultRestClient;
+import org.joda.time.DateTime;
+import com.microsoft.hsg.android.simplexml.HealthVaultApp;
+import com.microsoft.hsg.android.simplexml.client.IHealthVaultRestClient;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.request.BlobFormatSpec;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.request.ThingRequestGroup2;
+import com.microsoft.hsg.android.simplexml.things.thing.BlobInfo;
+import com.microsoft.hsg.android.simplexml.things.thing.BlobPayload;
+import com.microsoft.hsg.android.simplexml.things.thing.BlobPayloadItem;
+import com.microsoft.hsg.android.simplexml.things.thing.Thing2;
+import com.microsoft.hsg.android.simplexml.things.thing.ThingKey;
+import com.microsoft.hsg.android.simplexml.things.types.allergy.Allergy;
+import com.microsoft.hsg.android.simplexml.things.types.base.CodableValue;
+import com.microsoft.hsg.android.simplexml.things.types.types.PersonInfo;
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+
+import javax.inject.Inject;
+
+import dagger.Provides;
+
+import healthvault.client.implementation.MicrosoftHealthVaultRestApiImpl;
+import okhttp3.ConnectionPool;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+public class ActionPlanTest {
+	private static HealthVaultApp service;
+	private static HealthVaultClient hvClient;
+	private final String restHealthVaultUrl = "https://hvc-dev-khvwus01.westus2.cloudapp.azure.com/v3/";
```

Comment:
>v3/ [](start = 96, length = 3)

v3 shouldn't be needed anymore. Please remove v3 from the addresses. 

Also, since this is production code, this should be pointing to data.microsofthealth.net instead of the HVC dev cluster.